pkg/query/api/v1.go

```diff
@@ -95,20 +95,46 @@ type API struct {
 	queryEngine *promql.Engine
 	cfg         query.Config
 
-	now func() time.Time
+	instantQueryDuration prometheus.Histogram
+	rangeQueryDuration   prometheus.Histogram
+	now                  func() time.Time
 }
 
 // NewAPI returns an initialized API type.
 func NewAPI(
+	reg *prometheus.Registry,
 	qe *promql.Engine,
 	q promql.Queryable,
 	cfg query.Config,
 ) *API {
+	instantQueryDuration := prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name: "thanos_api_instant_query_duration_seconds",
```

Comment:
Probably we should lead with `thanos_<component/pkg>` in general, i.e. `thanos_query_api_instant_...`.